pkg/daemon/ceph/osd/kms/vault.go

```diff
@@ -0,0 +1,221 @@
+/*
+Copyright 2020 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kms
+
+import (
+	"io/ioutil"
+	"strings"
+
+	"github.com/hashicorp/vault/api"
+	"github.com/libopenstorage/secrets"
+	"github.com/libopenstorage/secrets/vault"
+	"github.com/pkg/errors"
+	"github.com/rook/rook/pkg/clusterd"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// EtcVaultDir is vault config dir
+	EtcVaultDir = "/etc/vault"
+)
+
+var (
+	vaultMandatoryConnectionDetails = []string{api.EnvVaultAddress}
+	vaultTLSConnectionDetails       = []string{api.EnvVaultCACert, api.EnvVaultClientCert, api.EnvVaultClientKey}
+)
+
+/* VAULT API INTERNAL VALUES
```

Comment:
Why list these here?

Reply:
Just a handy reminder while writing the code but it can go away. Let me know if you want to remove it.

Reply:
Since you link to the api doc, that seems better so this doesn't get out of date. Not a big deal though if you want to keep it.